tests/network/hotplug.go

```diff
@@ -405,6 +405,144 @@ var _ = SIGDescribe("nic-hotunplug", func() {
+			waitForUnplugIfaceApplyOnVMISpec(vmi, linuxBridgeNetworkName2)
+			vmi = verifyDynamicInterfaceChange(vmi, plugMethod)
+
+			Expect(libnet.InterfaceExists(vmi, guestNICName)).NotTo(Succeed(),
+				"unplugged interface should not exist in guest")
+
+			By("restarting the VM")
+			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).Restart(context.Background(), vm.Name, &v1.RestartOptions{})).To(Succeed())
+
+			By("wait for new VMI to start")
+			var newVMI *v1.VirtualMachineInstance
+			var err error
+			Eventually(func() v1.VirtualMachineInstancePhase {
+				newVMI, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+				if err != nil || vmi.UID == newVMI.UID {
+					return v1.VmPhaseUnset
+				}
+				return newVMI.Status.Phase
+			}, 90*time.Second, 1*time.Second).Should(Equal(v1.Running))
+			libwait.WaitUntilVMIReady(newVMI, console.LoginToAlpine)
+
+			By("verify unplugged interface does not exist in the new VMI spec")
+			newVMI = verifyDynamicInterfaceChange(newVMI, plugMethod)
+			Expect(libnet.InterfaceExists(newVMI, guestNICName)).ToNot(Succeed(),
+				"unplugged interface should not exist in guest after VM restart")
+		},
+			Entry("In place", decorators.InPlaceHotplugNICs, inPlace),
+			Entry("Migration based", decorators.MigrationBasedHotplugNICs, migrationBased),
+		)
+	})
+
+	Context("a stopped VM", func() {
+		var vm *v1.VirtualMachine
+		var vmi *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			By("create stopped VM")
+			opts := append(
+				libvmi.WithMasqueradeNetworking(),
+				libvmi.WithNetwork(libvmi.MultusNetwork(linuxBridgeNetworkName1, networkName)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(linuxBridgeNetworkName2, networkName)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(linuxBridgeNetworkName1)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(linuxBridgeNetworkName2)),
+			)
+			vmi = libvmi.NewAlpineWithTestTooling(opts...)
+			vm = tests.NewRandomVirtualMachine(vmi, false)
+
+			var err error
+			vm, err = kubevirt.Client().VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("cannot be subject to **hot** unplug, but will mutate the template.Spec on behalf of the user", func() {
+			removeIfaceOpts := &v1.RemoveInterfaceOptions{
+				Name: linuxBridgeNetworkName2,
+			}
+			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).RemoveInterface(context.Background(), vm.Name, removeIfaceOpts)).To(Succeed())
+
+			By("wait for the requested unplug interface spec update and have 'absent' state")
+			Eventually(func() []v1.Interface {
+				var err error
+				vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return vm.Spec.Template.Spec.Domain.Devices.Interfaces
+			}, 30*time.Second).Should(ConsistOf(
```

Comment:
I would not expect equality to work, especially when KMP is in the picture.
I would just invest in looking up for a specific interface in the status and check it has the absent on it, nothing more.

While on unit tests we can check with accuracy as we can stub/mock, on e2e tests it depends on many factors we have no control of. Lets limit the check.

Reply:
Done.